src/Features/Core/Portable/DesignerAttribute/AbstractDesignerAttributeIncrementalAnalyzer.cs

```diff
@@ -84,6 +79,35 @@ public DesignerAttributeDiscoveryService()
             if (!project.SupportsCompilation)
                 return;
 
+            // We're looking for "System.ComponentModel.DesignerCategoryAttribute", which is defined only in
+            // System.ComponentModel.Primitives.dll.  Avoid looking at any project that doesn't have a reference to that
+            // assembly.
+            if (!project.MetadataReferences.OfType<PortableExecutableReference>().Any(
+                    pe => pe.FilePath != null && pe.FilePath.EndsWith("System.ComponentModel.Primitives.dll", StringComparison.OrdinalIgnoreCase)))
+            {
+                return;
+            }
+
+            var compilation = await project.GetRequiredCompilationAsync(cancellationToken).ConfigureAwait(false);
+            var designerCategoryType = compilation.DesignerCategoryAttributeType();
+            if (designerCategoryType == null)
+                return;
```

Comment:
Consider adding an assert or non fatal or something -- if you keep the check for the assembly this should be in, this not being there might mean our earlier check is wrong and broken.

Reply:
yeah, removed the above, so this null check needs to stay.